test/services/user/services/AdminUsers.test.js

```diff
@@ -1667,4 +1667,93 @@ describe('AdminTeachersService', () => {
 		expect(deletedeStringType).to.not.be.undefined;
 		expect(deletedeStringType.firstName).to.equals('testDeleteTeacher');
 	});
+
+	it('can search the user data by firstName', async () => {
+		await testObjects.createTestRole({
+			name: 'studentListPerm',
+			permissions: ['STUDENT_LIST'],
+		});
+		const school = await testObjects.createTestSchool({
+			name: 'testSchool',
+		});
+		const testUSer = await testObjects.createTestUser({
+			roles: ['administrator'],
+			schoolId: school._id,
+			permissions: ['STUDENT_LIST'],
+		});
+		const student = await testObjects.createTestUser({
+			roles: ['student'],
+			schoolId: school._id,
+		});
+		const params = await testObjects.generateRequestParamsFromUser(testUSer);
+		params.query = {
+			...params.query,
+			searchQuery: student.firstName,
+		};
+
+		const result = await adminStudentsService.find(params);
+
+		expect(result.data).to.not.be.undefined;
+		expect(result.data[0].firstName).to.equal(student.firstName);
+		expect(result.data[0].lastName).to.equal(student.lastName);
+	});
+
+	it('can search the user data by lastName', async () => {
+		await testObjects.createTestRole({
+			name: 'studentListPerm',
+			permissions: ['STUDENT_LIST'],
+		});
+		const school = await testObjects.createTestSchool({
+			name: 'testSchool',
+		});
+		const testUSer = await testObjects.createTestUser({
+			roles: ['administrator'],
+			schoolId: school._id,
+			permissions: ['STUDENT_LIST'],
+		});
+		const student = await testObjects.createTestUser({
+			roles: ['student'],
+			schoolId: school._id,
+		});
+		const params = await testObjects.generateRequestParamsFromUser(testUSer);
+		params.query = {
+			...params.query,
+			searchQuery: student.lastName,
+		};
+		const result = await adminStudentsService.find(params);
+
+		expect(result.data).to.not.be.undefined;
+		expect(result.data[0].firstName).to.equal(student.firstName);
+		expect(result.data[0].lastName).to.equal(student.lastName);
+	});
+
+	it('can search the user data by firstName + lastName', async () => {
+		await testObjects.createTestRole({
+			name: 'studentListPerm',
+			permissions: ['STUDENT_LIST'],
+		});
+		const school = await testObjects.createTestSchool({
+			name: 'testSchool',
+		});
+		const testUSer = await testObjects.createTestUser({
+			roles: ['administrator'],
+			schoolId: school._id,
+			permissions: ['STUDENT_LIST'],
+		});
+		const student = await testObjects.createTestUser({
+			roles: ['student'],
+			schoolId: school._id,
+		});
+		const params = await testObjects.generateRequestParamsFromUser(testUSer);
+		params.query = {
+			...params.query,
+			searchQuery: `${student.firstName} ${student.lastName}`,
+		};
+
+		const result = await adminStudentsService.find(params);
+
+		expect(result.data).to.not.be.undefined;
+		expect(result.data[0].firstName).to.equal(student.firstName);
+		expect(result.data[0].lastName).to.equal(student.lastName);
+	});
```

Comment:
should create more user with simular names, so you realy know that you get the wanted user back at first. Set also a specific name for the user, so you can be sure its not only user with the default name